### PR TITLE
Feature: Implement comprehensive field validation for Kptfile (#4735)

### PR DESCRIPTION
**Pull request by adi-coderr**
### Description

**What changed:**
- Implement validation logic for unvalidated `KptFile` struct fields (`Upstream`, `UpstreamLock`, `Info`, and `Inventory`) in `api/kptfile/v1/validation.go`.
- Replace verbose struct literals with a concise `validationErr` helper, and inline the nested `Git` and `GitLock` validators into their parent structs, compacting the code without changing its time complexity.
- Refactor `validation_test.go` by merging four redundant test functions into a single `TestFieldValidation` table-driven test covering 20 essential edge cases.

**Why it's needed:**
- The `KptFile.Validate()` method previously only validated the `Pipeline` field, leaving a long-standing `TODO: validate other fields` on line 40.
- When authoring packages, users often edit the `Kptfile` by hand. Missing validation for mutually exclusive fields or invalid URL formats leads to confusing runtime errors much later in the pipeline rather than failing early during standard package validation. 

**How it works:**
- `Validate()` now explicitly calls `.validate()` on `Upstream`, `UpstreamLock`, `Info`, and `Inventory`.
- Pure in-memory string checks evaluate field consistency (e.g. `upstream.type` must match `upstream.git`, `info.licenseFile` cannot contain `..` traversal, and `Inventory` requires all fields or none).
- All validation short-circuits on the first error and returns the established `*ValidateError` type.

---

### Type of Change
- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [x] Refactor
- [ ] Documentation
- [x] Tests
- [ ] Other: ________

### Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed changes
- [x] Tests added/updated
- [ ] Documentation added/updated
- [x] All tests and gating checks pass

### AI Disclosure
- [x] I have used AI in the creation of this PR.

### CODE REVIEW

File: api/kptfile/v1/validation.go [hunk 1]
```diff
@@ -33,14 +34,137 @@ const (
 	kustomizationKind     = "Kustomization"
 )
 
+// validationErr creates a ValidateError with the given field, value, and reason.
+func validationErr(field, value, reason string) error {
+	return &ValidateError{Field: field, Value: value, Reason: reason}
+}
```
Comment (mozesl-nokia) [inline api/kptfile/v1/validation.go]: This function is redundant, just construct the error inline

File: api/kptfile/v1/validation_test.go [hunk 2]
```diff
@@ -693,3 +693,72 @@ func TestConfigRefValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestFieldValidation(t *testing.T) {
+	cases := []struct {
+		name  string
+		kf    KptFile
+		valid bool
+	}{
```
Comment (mozesl-nokia) [inline api/kptfile/v1/validation_test.go]: I believe we prefer using map[string]struct{}, where the key is the test name.

I think you should "explode" the case structs, so everything is on a newline like:
```go
"upstream: valid git": {
  kf: Kptfile{
    Upstream: &Upstream{
      Type: GitOrigin,
      Git: &Git{
        Repo: "https://github.com/kubernetes/examples.git",
        Ref: "main",
      },
      UpdateStrategy: ResourceMerge,
    },
  },
  valid: true
}
```

And you could probably split this test up based on "upstream", "lock", "info" and "inventory"

**Review by mozesl-nokia — COMMENTED** [review]
This is a more or less a more comprehensive version of https://github.com/kptdev/kpt/pull/4736, please contact each other about merging your PRs into one.

File: api/kptfile/v1/validation.go (hunk 1 above)
Comment (adi-coderr) [reply api/kptfile/v1/validation.go]: Updated

File: api/kptfile/v1/validation_test.go (hunk 2 above)
Comment (adi-coderr) [reply api/kptfile/v1/validation_test.go]: Updated

**Comment by liamfallon** [comment]
I checked locally and you have comprehensive unit test on new code, certainly over 80%

**Review by liamfallon — APPROVED** [review]

**Review by kushnaidu — APPROVED** [review]
